javalib/src/main/scala/java/util/Properties.scala

```diff
@@ -81,6 +120,275 @@ class Properties(protected val defaults: Properties)
       defaults.foreachAncestor(f)
   }
 
-  // def list(out: PrintStream): Unit
-  // def list(out: PrintWriter): Unit
+  private final val listStr = "-- listing properties --"
+
+  private def format(entry: ju.Map.Entry[AnyRef, AnyRef]): String = {
+    def format(s: String): String =
+      if (s.length > 40) s"${s.substring(0, 37)}..." else s
+
+    val key: String = entry.getKey().asInstanceOf[String]
+    val value: String = entry.getValue().asInstanceOf[String]
+
+    s"${key}=${format(value)}"
+  }
+
+  def list(out: PrintStream): Unit = {
+    out.println(listStr)
+    entrySet().scalaOps.foreach { entry => out.println(format(entry)) }
+  }
+
+  def list(out: PrintWriter): Unit = {
+    out.println(listStr)
+    entrySet().scalaOps.foreach { entry => out.println(format(entry)) }
+  }
+
+  private def loadImpl(reader: Reader): Unit = {
+    import java.util.regex._
+    val trailingBackspace = Pattern.compile("""(\\)+$""")
+    val br = new BufferedReader(reader)
+    val valBuf = new jl.StringBuilder()
+    var prevValueContinue = false
+    var isKeyParsed = false
+    var key: String = null
+    var line: String = null
+
+    while ({ line = br.readLine(); line != null }) {
+      var i: Int = -1
+      var ch: Char = Char.MinValue
+
+      def getNextChar(): Char = {
+        i += 1
+        // avoid out of bounds if value is empty
+        if (i < line.length())
+          line.charAt(i)
+        else
+          ch
+      }
+
+      def parseUnicodeEscape(): Char = {
+        val sb = new jl.StringBuilder()
+        var j = 0
+        while (j < 4) {
+          sb.append(line.charAt(i))
+          if (j < 3) {
+            // don't advance past the last char used
+            i += 1
+          }
+          j += 1
+        }
+        val ch = Integer.parseInt(sb.toString(), 16).toChar
+        ch
```

Comment:
Isn't this equivalent to
```scala
val ch = Integer.parseInt(sb.substring(i, i + 4).toChar
i += 3
ch
```
?

Reply:
I added a test to exercise this code directly to prove that I needed the `radix=16` here in Scala.js. On Native we read a unicode example from the file.